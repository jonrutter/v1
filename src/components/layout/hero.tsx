import React from 'react';
import clsx from 'clsx';

type Props = {
  short?: boolean;
};

/**
 * Renders a page's hero section.
 */
export const Hero: React.FC<Props> = ({ children, short }) => (
  <div
    className={clsx(
      'py-12 px-6 md:py-24 md:px-12 transition-all lg:flex lg:justify-center lg:items-center bg-white dark:bg-slate-900 text-base md:text-lg',
      !short && 'lg:min-h-[calc(100vh-5rem)]'
    )}
  >
    <div className="mx-auto max-w-prose">{children}</div>
  </div>
);
