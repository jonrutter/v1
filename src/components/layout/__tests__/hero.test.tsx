import React from 'react';
import { render, screen } from 'test-utils';

import { Hero } from '../hero';

const Component: React.FC<{ short?: boolean }> = ({ short = false }) => (
  <Hero title="Test Title" short={short}>
    <p>This is a test Hero component.</p>
  </Hero>
);

describe('Hero component', () => {
  it('renders the correct content', () => {
    render(<Component />);
    screen.getByText('Test Title');
    screen.getByText('This is a test Hero component.');
  });
});
