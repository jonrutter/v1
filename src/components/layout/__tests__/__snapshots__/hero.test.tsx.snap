// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component matches the snapshot 1`] = `
<div
  className="py-12 px-6 md:py-24 md:px-12 transition-all bg-white dark:bg-slate-900 text-base sm:text-lg lg:flex lg:items-center lg:min-h-[calc(100vh-5rem)]"
>
  <div
    className="max-w-site-sm md:max-w-site-md lg:max-w-site-lg mx-auto"
  >
    <div
      className="max-w-prose"
    >
      <div
        className="mb-6 md:mb-8 relative z-10 inline-block"
      >
        <h1
          className="transition-all font-heading font-black text-4xl md:text-5xl lg:text-6xl text-slate-900 dark:text-white mb-0"
        >
          Test Title
        </h1>
        <div
          className="shine opacity-50 dark:opacity-40"
        />
      </div>
      <p>
        This is a test Hero component.
      </p>
    </div>
  </div>
</div>
`;

exports[`Hero component matches the snapshot when short 1`] = `
<div
  className="py-12 px-6 md:py-24 md:px-12 transition-all bg-white dark:bg-slate-900 text-base sm:text-lg lg:flex lg:items-center"
>
  <div
    className="max-w-site-sm md:max-w-site-md lg:max-w-site-lg mx-auto"
  >
    <div
      className="max-w-prose"
    >
      <div
        className="mb-6 md:mb-8 relative z-10 inline-block"
      >
        <h1
          className="transition-all font-heading font-black text-4xl md:text-5xl lg:text-6xl text-slate-900 dark:text-white mb-0"
        >
          Test Title
        </h1>
        <div
          className="shine opacity-50 dark:opacity-40"
        />
      </div>
      <p>
        This is a test Hero component.
      </p>
    </div>
  </div>
</div>
`;
